### PR TITLE
Add Pipfile Analyzer

### PR DESCRIPTION
**Pull request by fcano**
#  Description of Change

Add a basic Pipfile Analyzer.

## Have test cases been added to cover the new functionality?

yes

### CODE REVIEW

File: core/src/main/java/org/owasp/dependencycheck/analyzer/PipfileAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                        d.addSoftwareIdentifier(new GenericIdentifier("pypi:" + identifiedPackage + "@" + identifiedVersion, Confidence.HIGH));
+                    }
+                    d.setPackagePath(String.format("%s:%s", identifiedPackage, identifiedVersion));
+                    d.setEcosystem(PythonDistributionAnalyzer.DEPENDENCY_ECOSYSTEM);
+                    final String filePath = String.format("%s:%s/%s", dependency.getFilePath(), identifiedPackage, identifiedVersion);
+                    d.setFilePath(filePath);
+                    d.setSha1sum(Checksum.getSHA1Checksum(filePath));
+                    d.setSha256sum(Checksum.getSHA256Checksum(filePath));
+                    d.setMd5sum(Checksum.getMD5Checksum(filePath));
+                    d.addEvidence(EvidenceType.PRODUCT, REQUIREMENTS, "product", identifiedPackage, Confidence.HIGHEST);
```
Comment (jeremylong) [inline core/src/main/java/org/owasp/dependencycheck/analyzer/PipfileAnalyzer.java:168]: Unfortunately - there is a requirement to have at least some vendor evidence. In a case like this where we don't have a vendor the recommendation is to just put the product name in as the vendor.

```suggestion
                    d.addEvidence(EvidenceType.VENDOR, REQUIREMENTS, "vendor", identifiedPackage, Confidence.HIGHEST);
                    d.addEvidence(EvidenceType.PRODUCT, REQUIREMENTS, "product", identifiedPackage, Confidence.HIGHEST);
```

**Review by jeremylong — CHANGES_REQUESTED** [review]
Thank you for such a complete PR! Just one minor suggestions - unless you can think of any other way to obtain vendor information from pip...

**Comment by fcano** [comment]
@jeremylong do you know why the test failed? Looking into the logs it does not seem a problem with the commit.

**Comment by jeremylong** [comment]
Apparently I broke the build.  If you `git rebase main` this should resolve itself.   Otherwise I know this builds and is correct - I can just merge...

**Comment by fcano** [comment]
The idea is to 'git rebase main' the branch 'add-pipfile-support' and drop the last commit (ab94853)?

**Comment by fcano** [comment]
I think I did it. Now, the PR does not have an evidence for the VENDOR. Is it mandatory? I don't see that line in the PipAnalyzer for example. It only has these two lines:

https://github.com/jeremylong/DependencyCheck/pull/2526/files#diff-13b45a953258e6e7d4327bc57ef26169R163-R164

File: core/src/main/java/org/owasp/dependencycheck/analyzer/PipfileAnalyzer.java (hunk 1 above)
Comment (jeremylong) [inline core/src/main/java/org/owasp/dependencycheck/analyzer/PipfileAnalyzer.java:168]:
```suggestion
                    d.addEvidence(EvidenceType.VENDOR, REQUIREMENTS, "vendor", identifiedPackage, Confidence.HIGHEST);
                    d.addEvidence(EvidenceType.PRODUCT, REQUIREMENTS, "product", identifiedPackage, Confidence.HIGHEST);
```

**Review by jeremylong — APPROVED** [review]

**Comment by jeremylong** [comment]
I just put the line adding the vendor evidence back. As to what happened - you likely accepted my change request in the github UI and did not pull the update back in before rebasing.

Seriously - that you for such a complete PR!

**Comment by fcano** [comment]
I'm sorry because I don't have a lot of experience with git and I got lost with the rebase. Now I see your commit in the pull request and the tests passing. Is everything ok now? Is there anything else that I have to do?

**Comment by jeremylong** [comment]
Yup - everything is good.  Thanks for the PR!

**Pull request merged**